### ISSUE TITLE
cnn autoencoder code

### ISSUE THREAD

**Issue by Phani123-tech** [Phani123-tech/cc#4]
"""CNN Autoencoder for ADAPT-SDN-Guard"""
import torch
import torch.nn as nn

class CNNAutoencoder(nn.Module):
    """CNN-based Autoencoder for anomaly detection"""
    def __init__(self, input_dim=48):
        super(CNNAutoencoder, self).__init__()
        self.input_dim = input_dim
        
        # Simple autoencoder architecture
        self.encoder = nn.Sequential(
            nn.Linear(input_dim, 64),
            nn.ReLU(),
            nn.Linear(64, 32),
            nn.ReLU(),
            nn.Linear(32, 16),
            nn.ReLU()
        )
        
        self.decoder = nn.Sequential(
            nn.Linear(16, 32),
            nn.ReLU(),
            nn.Linear(32, 64),
            nn.ReLU(),
            nn.Linear(64, input_dim),
            nn.Sigmoid()
        )
    
    def forward(self, x):
        encoded = self.encoder(x)
        decoded = self.decoder(encoded)
        return decoded
